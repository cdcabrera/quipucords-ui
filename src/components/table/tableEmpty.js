import React from 'react';
import PropTypes from 'prop-types';
import { Bullseye, EmptyState, EmptyStateVariant, EmptyStateIcon, EmptyStateBody, EmptyStateHeader,  } from '@patternfly/react-core';
import { SearchIcon } from '@patternfly/react-icons';
import { translate } from '../i18n/i18n';

const TableEmpty = ({ t }) => (
  <Bullseye>
    <EmptyState variant={EmptyStateVariant.sm}>
      <EmptyStateHeader titleText={<>{t('table.empty-state_title', 'No results found')}</>} icon={<EmptyStateIcon icon={SearchIcon} />} headingLevel="h2" />
      <EmptyStateBody>{t('table.empty-state_description', 'Clear all filters and try again.')}</EmptyStateBody>
    </EmptyState>
  </Bullseye>
);

TableEmpty.propTypes = {
  t: PropTypes.func
};

TableEmpty.defaultProps = {
  t: translate
};

export { TableEmpty as default, TableEmpty };
