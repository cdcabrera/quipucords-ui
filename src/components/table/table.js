import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useShallowCompareEffect } from 'react-use';
import _cloneDeep from 'lodash/cloneDeep';
import { Grid, GridItem } from '@patternfly/react-core';
import {
  ExpandableRowContent,
  SortByDirection,
  TableComposable,
  TableVariant,
  Tbody,
  Td,
  Th,
  Thead,
  Tr
} from '@patternfly/react-table';
import { TableEmpty } from './tableEmpty';
import { tableHelpers } from './tableHelpers';

const Table = ({
  ariaLabel,
  children,
  className,
  columnHeaders,
  isBorders,
  isHeader,
  rows,
  summary,
  variant,
  onSelect,
  onExpand
}) => {
  const [updatedHeaders, setUpdatedHeaders] = useState([]);
  const [updatedRows, setUpdatedRows] = useState([]);
  const [updatedHeaderSelectProps, setUpdatedHeaderSelectProps] = useState({});
  // const [updatedIsExpandableRow] = useState(false);
  const [updatedIsExpandableCell, setUpdatedIsExpandableCell] = useState(false);
  // const [updatedIsSortTable, setUpdatedIsSortTable] = useState(false);
  // const [updatedIsCollapsibleTable, setUpdatedIsCollapsibleTable] = useState(false);
  const [updatedIsSelectTable, setUpdatedIsSelectTable] = useState(false);

  // const [selectedRows, setSelectedRows] = useState({});
  // const [expandedRows, setExpandedRows] = useState([]);
  // const [expandedCells, setExpandedCells] = useState([]);
  // let isSelectTable = false;
  // let isExpandableCell = false;

  /**
   * Apply an onExpand handler.
   *
   * @param {object} params
   * @param {string} params.type
   * @param {number} params.rowIndex
   * @param {number} params.cellIndex
   */
  const onExpandTable = ({ type, rowIndex, cellIndex }) => {
    setUpdatedRows(value => {
      const updatedValue = [...value];

      if (type === 'compound') {
        const isCompoundExpanded = !updatedValue[rowIndex].cells[cellIndex].props.compoundExpand.isExpanded;

        updatedValue[rowIndex].cells = updatedValue[rowIndex].cells.map(({ props: cellProps, ...cell }) => {
          const updatedCompoundExpand = cellProps?.compoundExpand;

          if (updatedCompoundExpand) {
            updatedCompoundExpand.isExpanded = false;
          }

          return { ...cell, props: { ...cellProps, compoundExpand: updatedCompoundExpand } };
        });

        updatedValue[rowIndex].cells[cellIndex].props.compoundExpand.isExpanded = isCompoundExpanded;

        onExpand({
          type,
          rowIndex,
          cellIndex,
          isExpanded: isCompoundExpanded,
          cells: _cloneDeep(updatedValue[rowIndex].cells)
        });
      }

      return updatedValue;
    });
  };

  /**
   * Apply an onSelect handler.
   *
   * @param {object} params
   * @param {string} params.type
   * @param {number} params.rowIndex
   */
  const onSelectTable = ({ type, rowIndex }) => {
    if (type === 'all') {
      setUpdatedHeaderSelectProps(prevState => {
        const nextState = { ...prevState };
        const isSelected = !prevState.isSelected;

        nextState.isSelected = isSelected;

        setUpdatedRows(prevRowsState => {
          const nextRowsState = [...prevRowsState];
          nextRowsState.forEach(row => {
            const updatedRow = row;
            updatedRow.select.isSelected = isSelected;
          });

          onSelect({
            type,
            rowIndex,
            isSelected,
            rows: _cloneDeep(nextRowsState),
            cells: []
          });

          return nextRowsState;
        });

        return nextState;
      });

      // setUpdatedHeaderSelectProps(prevState => ({
      //         ...prevState,
      //         isSelected: !prevState.isSelected
      //       }));
      // if (type === 'all') {
      //           onSelect({
      //             type,
      //             rowIndex,
      //             // isSelected,
      //             rows: _cloneDeep(nextState[rowIndex])
      //           });
      //         }
    }

    if (type === 'row') {
      setUpdatedRows(prevState => {
        const nextState = [...prevState];
        const isSelected = !nextState[rowIndex].select.isSelected;

        nextState[rowIndex].select.isSelected = isSelected;
        const clonedRows = _cloneDeep(nextState);

        onSelect({
          type,
          rowIndex,
          isSelected,
          rows: clonedRows,
          cells: clonedRows[rowIndex].cells
        });

        return nextState;
      });
    }
  };

  useShallowCompareEffect(() => {
    console.log('>>>> update stuff');
    const {
      allRowsSelected,
      isSelectTable: parsedIsSelectTable,
      isExpandableCell: parsedIsExpandableCell,
      rows: parsedRows
    } = tableHelpers.tableRows({
      onExpand: typeof onExpand === 'function' && onExpandTable,
      onSelect: typeof onSelect === 'function' && onSelectTable,
      rows
    });

    const { columnHeaders: parsedColumnHeaders, headerSelectProps } = tableHelpers.tableHeader({
      columnHeaders,
      allRowsSelected,
      onSelect: typeof onSelect === 'function' && onSelectTable
      // isSelectTable: parsedIsSelectTable
    });

    console.log('header props >>>', headerSelectProps);
    // setUpdatedIsSortableTable
    // setUpdatedIsExpandableRow
    setUpdatedIsSelectTable(parsedIsSelectTable);
    setUpdatedIsExpandableCell(parsedIsExpandableCell);
    setUpdatedRows(parsedRows);
    setUpdatedHeaders(parsedColumnHeaders);
    setUpdatedHeaderSelectProps(headerSelectProps);
  }, [columnHeaders, onExpand, onExpandTable, onSelect, onSelectTable, rows]);

  // {updatedIsExpandableRow && <Td key="expand-th-cell" />}
  /**
   * Apply settings, return thead.
   *
   * @returns {React.ReactNode}
   */
  const renderHeader = () => {
    let selectProps = {};

    if (updatedHeaderSelectProps.select) {
      console.log('updated select props header >>>>>>', updatedHeaderSelectProps);
      selectProps = updatedHeaderSelectProps;
    }

    return (
      <Thead>
        <Tr>
          {updatedIsSelectTable && <Td key="select-th-cell" {...selectProps} />}
          {updatedHeaders.map(({ content, props, sort }) => (
            <Th key={tableHelpers.generateTableKey(content, 'th-cell')} sort={sort} {...props}>
              {content}
            </Th>
          ))}
        </Tr>
      </Thead>
    );
  };

  /**
   * Apply settings, return tbody.
   *
   * @returns {React.ReactNode}
   */
  const renderBody = () => {
    const BodyWrapper = (updatedIsExpandableCell && React.Fragment) || Tbody;

    return (
      <BodyWrapper>
        {updatedRows.map(({ cells, select }) => {
          const expandedCell = cells.find(cell => cell?.props?.compoundExpand?.isExpanded === true);
          const CellWrapper = (updatedIsExpandableCell && Tbody) || React.Fragment;
          const cellWrapperProps =
            (updatedIsExpandableCell && { isExpanded: expandedCell?.props?.compoundExpand?.isExpanded === true }) ||
            undefined;

          return (
            <CellWrapper key={tableHelpers.generateTableKey(cells, 'parent-row')} {...cellWrapperProps}>
              <Tr key={tableHelpers.generateTableKey(cells, 'row')}>
                {select && <Td key={tableHelpers.generateTableKey(cells, 'select-col')} select={select} />}
                {cells.map(({ content, isTHeader, props: cellProps }) => {
                  const WrapperCell = (isTHeader && Th) || Td;

                  return (
                    <WrapperCell key={tableHelpers.generateTableKey(content, 'cell')} {...cellProps}>
                      {content}
                    </WrapperCell>
                  );
                })}
              </Tr>
              {updatedIsExpandableCell && expandedCell && (
                <Tr isExpanded>
                  <Td colSpan={cells.length}>
                    <ExpandableRowContent>{expandedCell.expandedContent}</ExpandableRowContent>
                  </Td>
                </Tr>
              )}
            </CellWrapper>
          );
        })}
      </BodyWrapper>
    );
  };

  /**
   * Return empty results display.
   *
   * @returns {React.ReactNode}
   */
  const renderEmpty = () => children || <TableEmpty />;

  return (
    <Grid>
      <GridItem span={12}>
        {(updatedRows?.length && (
          <TableComposable
            aria-label={ariaLabel}
            borders={isBorders}
            className={className}
            summary={summary}
            variant={variant}
          >
            {isHeader && renderHeader()}
            {renderBody()}
          </TableComposable>
        )) ||
          renderEmpty()}
      </GridItem>
    </Grid>
  );
};

Table.propTypes = {
  ariaLabel: PropTypes.string,
  children: PropTypes.node,
  className: PropTypes.string,
  columnHeaders: PropTypes.arrayOf(
    PropTypes.oneOfType([
      PropTypes.func,
      PropTypes.node,
      PropTypes.shape({
        content: PropTypes.oneOfType([PropTypes.func, PropTypes.node]).isRequired,
        isSort: PropTypes.bool,
        onSort: PropTypes.func,
        sortDirection: PropTypes.oneOf([...Object.values(SortByDirection)])
      })
    ])
  ),
  isBorders: PropTypes.bool,
  isHeader: PropTypes.bool,
  // isSelected: PropTypes.bool, originally this was for selecting all rows... instead we make it a passive response in the "onSelect" user can... user should be setting every row they need
  // determine how to handle it... it'll be under type: "all"
  onExpand: PropTypes.func,
  onSelect: PropTypes.func,
  rows: PropTypes.arrayOf(
    PropTypes.shape({
      cells: PropTypes.arrayOf(
        PropTypes.oneOfType([
          PropTypes.func,
          PropTypes.node,
          PropTypes.instanceOf(Date),
          PropTypes.shape({
            content: PropTypes.oneOfType([PropTypes.func, PropTypes.node, PropTypes.instanceOf(Date)]).isRequired,
            isTHeader: PropTypes.bool,
            isExpanded: PropTypes.bool,
            expandedContent: PropTypes.oneOfType([PropTypes.node, PropTypes.func])
          })
        ])
      ),
      isDisabled: PropTypes.bool,
      isExpanded: PropTypes.bool,
      isSelected: PropTypes.bool,
      expandedContent: PropTypes.oneOfType([PropTypes.node, PropTypes.func])
    })
  ),
  summary: PropTypes.string,
  variant: PropTypes.oneOf([...Object.values(TableVariant)])
};

Table.defaultProps = {
  ariaLabel: null,
  children: null,
  className: 'quipucords-table',
  columnHeaders: [],
  isBorders: true,
  isHeader: false,
  onExpand: null,
  onSelect: null,
  rows: [],
  summary: null,
  variant: TableVariant.compact
};

export { Table as default, Table };
