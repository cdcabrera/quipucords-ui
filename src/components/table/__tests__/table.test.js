import React from 'react';
import { TableComposable, TableVariant, Th } from '@patternfly/react-table';
import { Table } from '../table';

describe('Table Component', () => {
  it('should render a basic component', async () => {
    const props = {
      isHeader: true,
      columnHeaders: ['lorem', 'ipsum', 'dolor', 'sit']
    };

    const component = await shallowHookComponent(<Table {...props} />);
    expect(component).toMatchSnapshot('basic');
  });

  it('should allow variations in table layout', async () => {
    const props = {
      isHeader: true,
      columnHeaders: ['lorem ipsum'],
      rows: [{ cells: ['dolor'] }, { cells: ['sit'] }]
    };

    const component = await mountHookComponent(<Table {...props} />);
    expect(component.find(TableComposable)).toMatchSnapshot('generated rows');

    component.setProps({
      isBorders: false,
      isHeader: false
    });
    expect(component.find(TableComposable)).toMatchSnapshot('borders and table header removed');

    component.setProps({
      ariaLabel: 'lorem ipsum aria-label',
      summary: 'lorem ipsum summary'
    });
    expect(component.find(TableComposable).props()).toMatchSnapshot('ariaLabel and summary');

    component.setProps({
      className: 'lorem-ipsum-class',
      variant: TableVariant.compact
    });
    expect(component.find(TableComposable).props()).toMatchSnapshot('className and variant');
  });

  it('should allow expandable row content', async () => {
    const mockOnExpand = jest.fn();
    const props = {
      onExpand: mockOnExpand,
      rows: [{ cells: ['dolor'], expandedContent: 'dolor sit expandable content' }, { cells: ['sit'] }]
    };

    const component = await mountHookComponent(<Table {...props} />);
    expect(component.find(TableComposable)).toMatchSnapshot('expandable row content');

    component.find('button').first().simulate('click');

    expect(mockOnExpand.mock.calls).toMatchSnapshot('expand row event');
    expect(component.find(TableComposable)).toMatchSnapshot('expanded row');
  });

  it('should allow expandable cell content', async () => {
    const mockOnExpand = jest.fn();
    const props = {
      onExpand: mockOnExpand,
      rows: [{ cells: [{ content: 'dolor', expandedContent: 'dolor expandable content' }] }, { cells: ['sit'] }]
    };

    const component = await mountHookComponent(<Table {...props} />);
    expect(component.find(TableComposable)).toMatchSnapshot('expandable cell content');

    component.find('button').first().simulate('click');

    expect(mockOnExpand.mock.calls).toMatchSnapshot('expand cell event');
    expect(component.find(TableComposable)).toMatchSnapshot('expanded cell');
  });

  it('should allow sortable content', async () => {
    const mockOnSort = jest.fn();
    const props = {
      isHeader: true,
      onSort: mockOnSort,
      columnHeaders: [{ content: 'lorem ipsum', isSort: true }],
      rows: [{ cells: ['dolor'] }, { cells: ['sit'] }]
    };

    const component = await mountHookComponent(<Table {...props} />);
    expect(component.find(Th).first()).toMatchSnapshot('sortable content');

    component.find('button').first().simulate('click');
    expect(component.find(Th).first()).toMatchSnapshot('sorted column, asc');

    component.find('button').first().simulate('click');
    expect(component.find(Th).first()).toMatchSnapshot('sorted column, desc');

    expect(mockOnSort.mock.calls).toMatchSnapshot('sort event');
  });

  it('should allow selectable row content', async () => {
    const mockOnSelect = jest.fn();
    const props = {
      isHeader: true,
      columnHeaders: ['lorem ipsum'],
      onSelect: mockOnSelect,
      rows: [{ cells: [{ content: 'dolor' }] }, { cells: ['sit'] }]
    };

    const component = await mountHookComponent(<Table {...props} />);
    expect(component.find('input')).toMatchSnapshot('select row content');

    component.find('input').first().simulate('change');

    expect(mockOnSelect.mock.calls).toMatchSnapshot('select row event');
    expect(component.find('input')).toMatchSnapshot('selected row');
  });

  it('should pass child components, nodes when there are no rows', async () => {
    const props = {
      isHeader: true,
      columnHeaders: ['lorem ipsum'],
      rows: []
    };

    const component = await shallowHookComponent(<Table {...props}>Loading...</Table>);
    expect(component).toMatchSnapshot('children');
  });
});
