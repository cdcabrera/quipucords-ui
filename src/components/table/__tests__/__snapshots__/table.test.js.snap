// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component should allow expandable cell content: expand cell event 1`] = `
[
  [
    {
      "cellIndex": 0,
      "data": {
        "data": {
          "hello": "world",
        },
      },
      "isExpanded": true,
      "rowIndex": 0,
      "type": "compound",
    },
  ],
]
`;

exports[`Table Component should allow expandable cell content: expandable cell content 1`] = `
<tr
  class="pf-v5-c-table__tr quipucords-table__tr"
  data-ouia-component-id="OUIA-Generated-TableRow-4"
  data-ouia-component-type="PF5/TableRow"
  data-ouia-safe="true"
>
  <td
    class="pf-v5-c-table__td  quipucords-table__td  pf-v5-c-table__compound-expansion-toggle"
    tabindex="-1"
  >
    <button
      aria-expanded="false"
      class="pf-v5-c-table__button"
      id="expand-toggle-undefined-undefined"
      type="button"
    >
      <span
        class="pf-v5-c-table__text"
      >
        dolor
      </span>
    </button>
  </td>
</tr>
`;

exports[`Table Component should allow expandable cell content: expanded cell 1`] = `
<td
  class="pf-v5-c-table__td quipucords-table__td quipucords-table__td-expand-expanded quipucords-table__td-expand-wrapper"
  colspan="1"
  tabindex="-1"
>
  <div
    class="quipucords-table__td-expand-content"
  >
    <div
      class="pf-v5-c-table__expandable-row-content"
    >
      dolor sit expandable content
    </div>
  </div>
</td>
`;

exports[`Table Component should allow expandable row content: expand row event 1`] = `
[
  [
    {
      "cellIndex": -1,
      "data": {
        "data": {
          "hello": "world",
        },
      },
      "isExpanded": true,
      "rowIndex": 0,
      "type": "row",
    },
  ],
]
`;

exports[`Table Component should allow expandable row content: expandable row content 1`] = `
<tr
  class="pf-v5-c-table__tr quipucords-table__tr"
  data-ouia-component-id="OUIA-Generated-TableRow-1"
  data-ouia-component-type="PF5/TableRow"
  data-ouia-safe="true"
>
  <td
    class="pf-v5-c-table__td quipucords-table__td quipucords-table__td-expand pf-v5-c-table__toggle"
    tabindex="-1"
  >
    <button
      aria-disabled="false"
      aria-expanded="false"
      aria-label="Details"
      aria-labelledby="simple-node0 expand-toggle0"
      class="pf-v5-c-button pf-m-plain"
      data-ouia-component-id="OUIA-Generated-Button-plain-1"
      data-ouia-component-type="PF5/Button"
      data-ouia-safe="true"
      id="expand-toggle0"
      type="button"
    >
      <div
        class="pf-v5-c-table__toggle-icon"
      >
        <svg
          aria-hidden="true"
          class="pf-v5-svg"
          fill="currentColor"
          height="1em"
          role="img"
          viewBox="0 0 320 512"
          width="1em"
        >
          <path
            d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
          />
        </svg>
      </div>
    </button>
  </td>
  <td
    class="pf-v5-c-table__td undefined quipucords-table__td "
    tabindex="-1"
  >
    dolor
  </td>
</tr>
`;

exports[`Table Component should allow expandable row content: expanded row 1`] = `
<td
  class="pf-v5-c-table__td quipucords-table__td quipucords-table__td-expand-expanded quipucords-table__td-expand-wrapper"
  colspan="2"
  tabindex="-1"
>
  <div
    class="quipucords-table__td-expand-content"
  >
    <div
      class="pf-v5-c-table__expandable-row-content"
    >
      dolor sit expandable content
    </div>
  </div>
</td>
`;

exports[`Table Component should allow selectable row content: select row content 1`] = `
<tr
  class="pf-v5-c-table__tr quipucords-table__tr"
  data-ouia-component-id="OUIA-Generated-TableRow-11"
  data-ouia-component-type="PF5/TableRow"
  data-ouia-safe="true"
>
  <td
    class="pf-v5-c-table__td quipucords-table__td quipucords-table__td-select pf-v5-c-table__check"
    tabindex="-1"
  >
    <label>
      <input
        aria-label="Select row 0"
        name="checkrow0"
        type="checkbox"
      />
    </label>
  </td>
  <td
    class="pf-v5-c-table__td  quipucords-table__td "
    tabindex="-1"
  >
    dolor
  </td>
</tr>
`;

exports[`Table Component should allow selectable row content: select row input 1`] = `
[
  [
    {
      "data": {
        "dataLabel": "testing",
      },
      "isSelected": true,
      "rowIndex": 0,
      "type": "row",
    },
  ],
]
`;

exports[`Table Component should allow sortable content: sort event 1`] = `
[
  [
    {
      "cellIndex": 0,
      "data": {},
      "direction": "asc",
    },
  ],
  [
    {
      "cellIndex": 0,
      "data": {},
      "direction": "desc",
    },
  ],
]
`;

exports[`Table Component should allow sortable content: sortable content 1`] = `
<th
  aria-sort="none"
  class="pf-v5-c-table__th quipucords-table__th pf-v5-c-table__sort"
  scope="col"
  tabindex="-1"
>
  <button
    class="pf-v5-c-table__button"
    type="button"
  >
    <div
      class="pf-v5-c-table__button-content"
    >
      <span
        class="pf-v5-c-table__text"
      >
        lorem ipsum
      </span>
      <span
        class="pf-v5-c-table__sort-indicator"
      >
        <svg
          aria-hidden="true"
          class="pf-v5-svg"
          fill="currentColor"
          height="1em"
          role="img"
          viewBox="0 0 256 512"
          width="1em"
        >
          <path
            d="M214.059 377.941H168V134.059h46.059c21.382 0 32.09-25.851 16.971-40.971L144.971 7.029c-9.373-9.373-24.568-9.373-33.941 0L24.971 93.088c-15.119 15.119-4.411 40.971 16.971 40.971H88v243.882H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.568 9.373 33.941 0l86.059-86.059c15.12-15.119 4.412-40.971-16.97-40.971z"
          />
        </svg>
      </span>
    </div>
  </button>
</th>
`;

exports[`Table Component should allow sortable content: sorted column, asc 1`] = `
<th
  aria-sort="ascending"
  class="pf-v5-c-table__th quipucords-table__th pf-v5-c-table__sort pf-m-selected"
  scope="col"
  tabindex="-1"
>
  <button
    class="pf-v5-c-table__button"
    type="button"
  >
    <div
      class="pf-v5-c-table__button-content"
    >
      <span
        class="pf-v5-c-table__text"
      >
        lorem ipsum
      </span>
      <span
        class="pf-v5-c-table__sort-indicator"
      >
        <svg
          aria-hidden="true"
          class="pf-v5-svg"
          fill="currentColor"
          height="1em"
          role="img"
          viewBox="0 0 256 512"
          width="1em"
        >
          <path
            d="M88 166.059V468c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12V166.059h46.059c21.382 0 32.09-25.851 16.971-40.971l-86.059-86.059c-9.373-9.373-24.569-9.373-33.941 0l-86.059 86.059c-15.119 15.119-4.411 40.971 16.971 40.971H88z"
          />
        </svg>
      </span>
    </div>
  </button>
</th>
`;

exports[`Table Component should allow sortable content: sorted column, desc 1`] = `
<th
  aria-sort="descending"
  class="pf-v5-c-table__th quipucords-table__th pf-v5-c-table__sort pf-m-selected"
  scope="col"
  tabindex="-1"
>
  <button
    class="pf-v5-c-table__button"
    type="button"
  >
    <div
      class="pf-v5-c-table__button-content"
    >
      <span
        class="pf-v5-c-table__text"
      >
        lorem ipsum
      </span>
      <span
        class="pf-v5-c-table__sort-indicator"
      >
        <svg
          aria-hidden="true"
          class="pf-v5-svg"
          fill="currentColor"
          height="1em"
          role="img"
          viewBox="0 0 256 512"
          width="1em"
        >
          <path
            d="M168 345.941V44c0-6.627-5.373-12-12-12h-56c-6.627 0-12 5.373-12 12v301.941H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.569 9.373 33.941 0l86.059-86.059c15.119-15.119 4.411-40.971-16.971-40.971H168z"
          />
        </svg>
      </span>
    </div>
  </button>
</th>
`;

exports[`Table Component should allow variations in table layout: ariaLabel and summary 1`] = `
<Grid>
  <GridItem
    span={12}
  >
    <Table
      actionsMenuAppendTo="inline"
      aria-label="lorem ipsum aria-label"
      borders={true}
      canCollapseAll={false}
      canSelectAll={true}
      canSortFavorites={true}
      className="quipucords-table "
      collapseAllAriaLabel=""
      contentId="expanded-content"
      dropdownDirection="down"
      dropdownPosition="right"
      expandId="expandable-toggle"
      gridBreakPoint="grid-md"
      isHeaderSelectDisabled={false}
      isNested={false}
      isStickyHeader={false}
      isStriped={false}
      isTreeTable={false}
      ouiaSafe={true}
      role="grid"
      rowLabeledBy="simple-node"
      selectVariant="checkbox"
      summary="lorem ipsum summary"
      variant="compact"
    >
      <Thead>
        <Tr
          className="quipucords-table__tr"
        >
          <Th
            className="quipucords-table__th"
          >
            lorem ipsum
          </Th>
        </Tr>
      </Thead>
      <Tbody>
        <React.Fragment>
          <Tr
            className="quipucords-table__tr"
          >
            <Td
              className="undefined quipucords-table__td "
            >
              dolor
            </Td>
          </Tr>
        </React.Fragment>
        <React.Fragment>
          <Tr
            className="quipucords-table__tr"
          >
            <Td
              className="undefined quipucords-table__td "
            >
              sit
            </Td>
          </Tr>
        </React.Fragment>
      </Tbody>
    </Table>
  </GridItem>
</Grid>
`;

exports[`Table Component should allow variations in table layout: borders and table header removed 1`] = `
<Grid>
  <GridItem
    span={12}
  >
    <Table
      actionsMenuAppendTo="inline"
      aria-label={null}
      borders={false}
      canCollapseAll={false}
      canSelectAll={true}
      canSortFavorites={true}
      className="quipucords-table "
      collapseAllAriaLabel=""
      contentId="expanded-content"
      dropdownDirection="down"
      dropdownPosition="right"
      expandId="expandable-toggle"
      gridBreakPoint="grid-md"
      isHeaderSelectDisabled={false}
      isNested={false}
      isStickyHeader={false}
      isStriped={false}
      isTreeTable={false}
      ouiaSafe={true}
      role="grid"
      rowLabeledBy="simple-node"
      selectVariant="checkbox"
      summary={null}
      variant="compact"
    >
      <Tbody>
        <React.Fragment>
          <Tr
            className="quipucords-table__tr"
          >
            <Td
              className="undefined quipucords-table__td "
            >
              dolor
            </Td>
          </Tr>
        </React.Fragment>
        <React.Fragment>
          <Tr
            className="quipucords-table__tr"
          >
            <Td
              className="undefined quipucords-table__td "
            >
              sit
            </Td>
          </Tr>
        </React.Fragment>
      </Tbody>
    </Table>
  </GridItem>
</Grid>
`;

exports[`Table Component should allow variations in table layout: className and variant 1`] = `
<Grid>
  <GridItem
    span={12}
  >
    <Table
      actionsMenuAppendTo="inline"
      aria-label={null}
      borders={true}
      canCollapseAll={false}
      canSelectAll={true}
      canSortFavorites={true}
      className="quipucords-table lorem-ipsum-class"
      collapseAllAriaLabel=""
      contentId="expanded-content"
      dropdownDirection="down"
      dropdownPosition="right"
      expandId="expandable-toggle"
      gridBreakPoint="grid-md"
      isHeaderSelectDisabled={false}
      isNested={false}
      isStickyHeader={false}
      isStriped={false}
      isTreeTable={false}
      ouiaSafe={true}
      role="grid"
      rowLabeledBy="simple-node"
      selectVariant="checkbox"
      summary={null}
      variant="compact"
    >
      <Thead>
        <Tr
          className="quipucords-table__tr"
        >
          <Th
            className="quipucords-table__th"
          >
            lorem ipsum
          </Th>
        </Tr>
      </Thead>
      <Tbody>
        <React.Fragment>
          <Tr
            className="quipucords-table__tr"
          >
            <Td
              className="undefined quipucords-table__td "
            >
              dolor
            </Td>
          </Tr>
        </React.Fragment>
        <React.Fragment>
          <Tr
            className="quipucords-table__tr"
          >
            <Td
              className="undefined quipucords-table__td "
            >
              sit
            </Td>
          </Tr>
        </React.Fragment>
      </Tbody>
    </Table>
  </GridItem>
</Grid>
`;

exports[`Table Component should allow variations in table layout: generated rows 1`] = `
<Grid>
  <GridItem
    span={12}
  >
    <Table
      actionsMenuAppendTo="inline"
      aria-label={null}
      borders={true}
      canCollapseAll={false}
      canSelectAll={true}
      canSortFavorites={true}
      className="quipucords-table "
      collapseAllAriaLabel=""
      contentId="expanded-content"
      dropdownDirection="down"
      dropdownPosition="right"
      expandId="expandable-toggle"
      gridBreakPoint="grid-md"
      isHeaderSelectDisabled={false}
      isNested={false}
      isStickyHeader={false}
      isStriped={false}
      isTreeTable={false}
      ouiaSafe={true}
      role="grid"
      rowLabeledBy="simple-node"
      selectVariant="checkbox"
      summary={null}
      variant="compact"
    >
      <Thead>
        <Tr
          className="quipucords-table__tr"
        >
          <Th
            className="quipucords-table__th"
          >
            lorem ipsum
          </Th>
        </Tr>
      </Thead>
      <Tbody>
        <React.Fragment>
          <Tr
            className="quipucords-table__tr"
          >
            <Td
              className="undefined quipucords-table__td "
            >
              dolor
            </Td>
          </Tr>
        </React.Fragment>
        <React.Fragment>
          <Tr
            className="quipucords-table__tr"
          >
            <Td
              className="undefined quipucords-table__td "
            >
              sit
            </Td>
          </Tr>
        </React.Fragment>
      </Tbody>
    </Table>
  </GridItem>
</Grid>
`;

exports[`Table Component should pass child components, nodes when there are no rows: children 1`] = `
<Grid>
  <GridItem
    span={12}
  >
    Loading...
  </GridItem>
</Grid>
`;

exports[`Table Component should render a basic component: basic 1`] = `
<Grid>
  <GridItem
    span={12}
  >
    <TableEmpty
      t={[Function]}
    />
  </GridItem>
</Grid>
`;
